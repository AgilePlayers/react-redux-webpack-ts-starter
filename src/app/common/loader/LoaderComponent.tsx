import React from 'react';

import './loader.scss';

const LoaderComponent: React.SFC = () => (
  <>
    <h1>Loader</h1>
    <div className="loader">Data is loading...</div>
    <div className="loader-spinner" />
  </>
);

export default LoaderComponent;
