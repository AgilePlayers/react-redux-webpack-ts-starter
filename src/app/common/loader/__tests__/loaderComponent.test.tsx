import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import LoaderComponent from '../LoaderComponent';

describe('common / loader / LoaderComponent', () => {
  let component: ShallowWrapper;

  beforeEach(() => {
    component = shallow(<LoaderComponent />);
  });

  test('should render', () => {
    expect(component.length).toEqual(1);
  });

  test('should match snapshot', () => {
    expect(component).toMatchSnapshot();
  });

  test('should contains element with "loader" class', () => {
    expect(component.find('div.loader').length).toEqual(1);
    expect(component.find('div.loader-spinner').length).toEqual(1);
  });
});
