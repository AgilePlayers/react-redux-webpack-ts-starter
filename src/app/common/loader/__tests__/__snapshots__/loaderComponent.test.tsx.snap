// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`common / loader / LoaderComponent should match snapshot 1`] = `ShallowWrapper {}`;
